src/beignet/_minimum_spanning_tree.py

```diff
@@ -0,0 +1,150 @@
+import torch
+from torch import Tensor
+
+
+def minimum_spanning_tree(
+    num_nodes: int,
+    edge_index: Tensor,
+    edge_weight: Tensor,
+) -> tuple[Tensor, Tensor]:
```

Comment:
This API sucks. It should be as simple as:

`kruskal(input: Tensor) -> Tensor`

where the input is a (N, N) matrix or (..., N, N) matrices representing one or more undirected graphs over N nodes and output is the (N, N) or (..., N, N) compressed-sparse representation of the undirected minimum spanning tree over the input.

This operator should use undirected graphs as input _and_ output. That is, if graph[i, j] and graph[j, i] are both zero, then nodes i and j do not have an edge connecting them. If either is non-zero, then the two are connected by the minimum non-zero value of the two.

Users should input sparse matrices, if possible, to avoid precision issues.

Finally, you should be operating on, and only operating on, sparse CSR tensors from `torch.sparse`!